### PR TITLE
[PM-7693] Remove cookie from Duo connector

### PR DESCRIPTION
**Pull request by ike-kottlowski**
## 🎟️ Tracking

[PM-7693](https://bitwarden.atlassian.net/browse/PM-7693)

## 📔 Objective

Implement Translation Service **not** through cookie. We are removing the cookie implementation to reduce attack surface.

### Desktop
https://github.com/bitwarden/clients/assets/137194738/d94733df-f669-4059-b10f-611f0652064d

### Browser

https://github.com/bitwarden/clients/assets/137194738/9e3f78c6-91f5-4508-857b-aeff33fc3553

### Web

https://github.com/bitwarden/clients/assets/137194738/100ba80e-5c86-46d5-a850-1e28eec3e338

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-7693]: https://bitwarden.atlassian.net/browse/PM-7693?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/desktop/src/auth/two-factor.component.ts [hunk 1]
```diff
@@ -149,22 +149,14 @@ export class TwoFactorComponent extends BaseTwoFactorComponent {
...
       env.getWebVaultUrl() +
       "/duo-redirect-connector.html" +
-      "?duoFramelessUrl=" +
-      encodeURIComponent(this.duoFramelessUrl) +
-      "&handOffMessage=" +
-      encodeURIComponent(JSON.stringify(duoHandOffMessage));
+      "?locale=" +
+      encodeURIComponent(this.i18nService.translationLocale) +
+      "&duoFramelessUrl=" +
+      encodeURIComponent(this.duoFramelessUrl);
```
[the remark this answers is not in the document]
Comment (ike-kottlowski) [reply apps/desktop/src/auth/two-factor.component.ts]: I was actually able to remove all the override methods in the clients and use the base `two-factor.component.ts` to implement the `launchDuoUri()` method.

Good catch thanks!

File: apps/browser/src/auth/popup/two-factor.component.ts [hunk 2]
```diff
@@ -224,4 +224,24 @@ export class TwoFactorComponent extends BaseTwoFactorComponent {
         });
     }
   }
+
+  override async launchDuoFrameless() {
+    const duoHandOffMessage = {
+      title: this.i18nService.t("youSuccessfullyLoggedIn"),
+      message: this.i18nService.t("youMayCloseThisWindow"),
+      isCountdown: false,
+    };
```
Comment (rr-bw) [inline apps/browser/src/auth/popup/two-factor.component.ts:233]: Question: how does this get parsed if we removed the [relevant code](https://github.com/bitwarden/clients/pull/9699/files#diff-b2c40447e255d4d8cd2deb93e6de05db316c891d548c853bfd95712f34a66b3fL122-L124) from `duo-redirect.ts`?

File: apps/web/src/connectors/duo-redirect.ts [hunk 3]
```diff
@@ -42,41 +46,20 @@ window.addEventListener("load", () => {
...
- *  message: "This window will automatically close in 5 seconds",
- *  buttonText: "Close",
- *  isCountdown: true
- * };
- *
- * document.cookie = `duoHandOffMessage=${encodeURIComponent(JSON.stringify(duoHandOffMessage))};SameSite=strict`;
- *
- * ```
- *
  * The `title`, `message`, and `buttonText` properties will be used to create the
```
Comment (rr-bw) [inline apps/web/src/connectors/duo-redirect.ts:63]: Do you want to remove / edit this large comment comment to reflect changes?

File: apps/browser/src/auth/popup/two-factor.component.ts (hunk 2 above)
Comment (ike-kottlowski) [reply apps/browser/src/auth/popup/two-factor.component.ts:233]: This is to support a case where a user is targeting a self-hosted environment that still requires the `duoHandOffMessage` in order to show proper messaging in the `duo-redirect.ts`.

**Review by rr-bw — APPROVED** [review]

**Pull request merged**